ml_model_metadata_node.py

```diff
@@ -113,12 +119,14 @@ def task_callback(user_input, node_status, ml_model_metadata):
     # Callback implementation here
     global graph
     print (f"Received Task: {user_input.task_id().problem_id()},{user_input.task_id().iteration_id()}")
+    client = Client(host='http://localhost:11434')
 
     try:
         extra_data_bytes = user_input.extra_data()
         extra_data_str = ''.join(chr(b) for b in extra_data_bytes)
         extra_data_dict = json.loads(extra_data_str)
         accumulated_data = {}
+        dataset_metadata = {}
```

Comment:
Move `dataset_metadata = {}` before the try of extra_data so in case the retrieve of extra_data gave an error, for example in tests extra_data is not send, `dataset_metadata` is still defined.
This change should fix the error on tests of SuatainML_Library.

```suggestion
    client = Client(host='http://localhost:11434')
    dataset_metadata = {}

    try:
        extra_data_bytes = user_input.extra_data()
        extra_data_str = ''.join(chr(b) for b in extra_data_bytes)
        extra_data_dict = json.loads(extra_data_str)
        accumulated_data = {}
```